From 9ec7b053f9c693e8fa49bcde26a971c8a8440951 Mon Sep 17 00:00:00 2001
From: serlk <22211586+Serlk@users.noreply.github.com>
Date: Sun, 18 Oct 2026 14:56:27 +0300
Subject: [PATCH] inputflinger: allow disabling input devices via idc

On some devices an external Daemon consumes input events and these
shouldn't be reported to apps (Lindroid).

Forward-port of LMODroid change 12936 to Android 16 (getBool returns optional).

Change-Id: Iffedb3061361ee120563c1f5e39a14faf5a2f83c
---
 services/inputflinger/reader/EventHub.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/services/inputflinger/reader/EventHub.cpp b/services/inputflinger/reader/EventHub.cpp
index 56dd2e9..f2c47f2 100644
--- a/services/inputflinger/reader/EventHub.cpp
+++ b/services/inputflinger/reader/EventHub.cpp
@@ -2465,6 +2465,13 @@ void EventHub::openDeviceLocked(const std::string& devicePath) {
     // Obtain the associated device, if any.
     device->associatedDevice = obtainAssociatedDeviceLocked(devicePath, device->configuration);
 
+    // Disable device if device config property set
+    if (device->configuration &&
+        device->configuration->getBool("device.disabled").value_or(false)) {
+        device->disable();
+        ALOGV("Disabling device with id %d\n", device->id);
+    }
+
     // Figure out the kinds of events the device reports.
     device->readDeviceBitMask(EVIOCGBIT(EV_KEY, 0), device->keyBitmask);
     device->readDeviceBitMask(EVIOCGBIT(EV_ABS, 0), device->absBitmask);
-- 
2.56.0

